tests/ui/ptr_eq.fixed

```diff
@@ -0,0 +1,23 @@
+// run-rustfix
+#![warn(clippy::ptr_eq)]
+
+fn main() {
+    let a = &[1, 2, 3];
+    let b = &[1, 2, 3];
+
+    let _ = std::ptr::eq(a, b);
+    let _ = std::ptr::eq(a, b);
```

Comment:
Does this compile? Doesn't this has to be 
```suggestion
    let _ = std::ptr::eq(a as *const _, b as *const _);
```

Reply:
Yes, it does compile.